retrieval/discovery/ec2.go

```diff
@@ -72,20 +72,22 @@ func init() {
 type EC2Discovery struct {
 	aws      *aws.Config
 	interval time.Duration
+	profile  string
 	port     int
 }
 
 // NewEC2Discovery returns a new EC2Discovery which periodically refreshes its targets.
 func NewEC2Discovery(conf *config.EC2SDConfig) *EC2Discovery {
 	creds := credentials.NewStaticCredentials(conf.AccessKey, conf.SecretKey, "")
 	if conf.AccessKey == "" && conf.SecretKey == "" {
-		creds = defaults.DefaultChainCredentials
+		creds = nil
```

Comment:
Does this have anything to do with the changes or does it relate to some changes in the updated vendoring?